app/controllers/newflow/login_controller.rb

```diff
@@ -24,13 +24,14 @@ def login
           redirect_back # back to `r`edirect parameter. See `before_action :save_redirect`.
         },
         failure: lambda {
-          save_login_failed_email(@handler_result.outputs.email)
+          email = @handler_result.outputs.email
+          save_login_failed_email(email)
 
           code = @handler_result.errors.first.code
           case code
           when :cannot_find_user, :multiple_users, :incorrect_password, :too_many_login_attempts
             user = @handler_result.outputs.user
-            security_log(:sign_in_failed, { reason: code, user: user }) # also store email?
+            security_log(:sign_in_failed, { reason: code, email: email })
```

Comment:
what's the reason behind changing this to email? it concerns me for GDPR reasons, if we get a delete request, we are not supposed to remove all user PII data from our systems, user would be more abstract.

Reply:
1) `user` was actually `nil` there but if it wasn't, then 2) the function `security_log` already captures whatever user is logged in as `current_user`. So that part of that line wasn't doing anything. With `email` now we actually track the attempt to log in with a specific email.